confection/__init__.py

```diff
@@ -946,6 +1024,8 @@ def _update_from_parsed(
             # Check numpy first, just in case. Use stringified type so that numpy dependency can be ditched.
             elif str(type(value)) == "<class 'numpy.ndarray'>":
                 final[key] = value
+            elif isinstance(value, BaseModel) and isinstance(final[key], BaseModel):
```

Comment:
This case can occur from the above change to pass through resolved pydantic models